zerver/tests/test_message_edit.py

```diff
@@ -2216,6 +2224,23 @@ def test_mark_topic_as_resolved(self) -> None:
             f"@_**Iago|{admin_user.id}** has marked this topic as resolved.",
         )
 
+        # Check topic resolved notification message is only unread for participants.
+        assert (
+            UserMessage.objects.filter(
+                user_profile__in=[admin_user, hamlet, aaron], message__id=messages[2].id
+            )
+            .extra(where=[UserMessage.where_unread()])
+            .count()
+            == 3
+        )
+
+        assert (
+            UserMessage.objects.filter(user_profile=cordelia, message__id=messages[2].id)
+            .extra(where=[UserMessage.where_unread()])
+            .count()
+            == 0
+        )
```

Comment:
Can you extend the test to have an additional user, maybe Prospero, add an emoji reaction to a message in the topic, so that we test that part of the conditional?

Reply:
Yeah, I did it [here](https://github.com/zulip/zulip/pull/20977/files/acfa21313c7e3f50aa807e8ba90d265326d0dfc7#diff-3a0986e298aade053cf2a8360758507bc4fed8c3d1fe341724e6c7fd0dd720a6R2186) with aaron.